pkg/devfile/parser/data/interface.go

```diff
@@ -26,26 +26,30 @@ type DevfileData interface {
 	GetComponents(common.DevfileOptions) ([]v1.Component, error)
 	AddComponents(components []v1.Component) error
 	UpdateComponent(component v1.Component)
+	DeleteComponent(name string) error
 
 	// project related methods
 	GetProjects(common.DevfileOptions) ([]v1.Project, error)
 	AddProjects(projects []v1.Project) error
 	UpdateProject(project v1.Project)
+	DeleteProject(name string) error
 
 	// starter projects related commands
 	GetStarterProjects(common.DevfileOptions) ([]v1.StarterProject, error)
 	AddStarterProjects(projects []v1.StarterProject) error
 	UpdateStarterProject(project v1.StarterProject)
+	DeleteStarterProject(name string) error
 
 	// command related methods
 	GetCommands(common.DevfileOptions) ([]v1.Command, error)
-	AddCommands(commands ...v1.Command) error
+	AddCommands(commands []v1.Command) error
 	UpdateCommand(command v1.Command)
+	DeleteCommand(id string) error
 
-	// volume related methods
-	AddVolume(volume v1.Component, path string) error
-	DeleteVolume(name string) error
```

Comment:
This is removing existing functions that other tools might be already using. This is a breaking change.

Reply:
you're right